src/main/java/edu/indiana/dlib/amppd/model/BatchSupplementFile.java

```diff
@@ -24,13 +26,13 @@
 
 	@Id
     @GeneratedValue(strategy=GenerationType.AUTO)
-	private Long  id;
+	private Long id;
 	private int supplementNum;
-	
 	private String supplementFilename;	
 	private String supplementName;
 	private String supplementDescription;
-	
+
+	@JsonIgnore
 	@ManyToOne
 	private BatchFile batchFile;
 	
```

Comment:
I ran into similar infinite loop issues with serialization in several cases.  One situation is caused by Lombok toString, equals and hashCode methods not handling such bi-direction reference gracefully, thus causing the loop. the solution I found was to set (callSuper=true, onlyExplicitlyIncluded=true), which exclude fields other than annotated. One example of this case is with Item and Collection. 

Another situation is, if i remember correctly, related to @ManyToMany relations (see Primaryfile and Bundle). the solution was to add @JsonBackReference on one side. 

I'm not sure which scenario you ran into the infinite loop. Using @JsonIgnore solves the problem, but I wonder if there is a side effect  - for ex, will the ID of the reference be excluded in the response, which may be undesirable? I wonder whether any of the solutions i used before, or other light-weight solution will resolve the issue?

Reply:
I changed most of the references to JsonBackReference.  For some in the batch, however, I left those as JsonIgnore (Unit, etc) where it seemed to make sense.